src/main/java/com/productdock/library/inventory/adapter/out/mongo/InventoryRecordsRepository.java

```diff
@@ -0,0 +1,35 @@
+package com.productdock.library.inventory.adapter.out.mongo;
+
+import com.productdock.library.inventory.adapter.out.mongo.mapper.InventoryRecordMapper;
+import com.productdock.library.inventory.application.port.out.persistence.InventoryRecordsPersistenceOutPort;
+import com.productdock.library.inventory.domain.Inventory;
+import com.productdock.library.inventory.domain.exception.InventoryException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Slf4j
+@AllArgsConstructor
+public class InventoryRecordsRepository implements InventoryRecordsPersistenceOutPort {
+
+    private InventoryRecordEntityRepository inventoryRecordEntityRepository;
+    private InventoryRecordMapper inventoryRecordMapper;
+
+    @Override
+    public Inventory getInventoryFrom(String bookId) {
```

Comment:
Rename method to `findByBookId`

Reply:
Return `Optional<Inventory>` instead of throwing exception. Repository should not throw exception in `findBy*` method.
If necessary, throw exception in service layer.